### PR TITLE
feat(ui): add html to markdown conversion and vice versa in block editor

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes:

I worked on adding support for the following things in the block editor
- HTML to markdown conversion and vice versa
- Mention link should have the entity type and absolute URL
- Custom Node for diff view
- Extension for image
- Added field support in entity link builder util

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/BlockEditor/Extensions/image/index.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      setImage: (options: {
+        src: string;
+        alt?: string;
+        title?: string;
+      }) => ReturnType;
+    };
+  }
+}
+
+export const inputRegex =
```
Comment (karanh37) [inline openmetadata-ui/src/main/resources/ui/src/components/BlockEditor/Extensions/image/index.ts:36]: Move this to constants

**Review by karanh37 — CHANGES_REQUESTED** [review]

**Review by karanh37 — DISMISSED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/BlockEditor/Extensions/mention/mentionSuggestions.ts [hunk 2]
```diff
@@ -15,8 +15,9 @@ import tippy, { Instance, Props } from 'tippy.js';
 
 import { SuggestionKeyDownProps, SuggestionProps } from '@tiptap/suggestion';
 import { WILD_CARD_CHAR } from 'constants/char.constants';
-import { getTeamAndUserDetailsPath, getUserPath } from 'constants/constants';
+import { entityUrlMap } from 'constants/Feeds.constants';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/BlockEditor/Extensions/mention/mentionSuggestions.ts:18]: const names should be PASCAL_CASE

File: openmetadata-ui/src/main/resources/ui/src/components/BlockEditor/Extensions/mention/mentionSuggestions.ts [hunk 3]
```diff
@@ -46,10 +47,10 @@ export const mentionSuggestion = () => ({
         name: hit._source.name,
         label: hit._source.displayName,
         fqn: hit._source.fullyQualifiedName,
-        href:
-          hit._source.entityType === 'user'
-            ? getUserPath(hit._source.fullyQualifiedName ?? '')
-            : getTeamAndUserDetailsPath(hit._source.fullyQualifiedName ?? ''),
+        href: buildMentionLink(
+          entityUrlMap[hit._source.entityType as keyof typeof entityUrlMap],
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/BlockEditor/Extensions/mention/mentionSuggestions.ts:51]: Can we extract it as type and use that type?
```
as keyof typeof entityUrlMap
```

File: openmetadata-ui/src/main/resources/ui/src/utils/FeedElementUtils.tsx [hunk 4]
```diff
@@ -58,19 +55,3 @@ export const getFieldThreadElement = (
...
-    const name = owner.name;
-    const displayName = owner.displayName;
-    const entityType = owner.type;
-    const mention = `<a href=${`/${
-      entityUrlMap[entityType as keyof typeof entityUrlMap]
-    }/${name}`}>@${displayName}</a>`;
-
-    return `${mention} ${message}`;
-  }
-};
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/FeedElementUtils.tsx:76]: Removed this as we are not using it.

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**